usr/share/lib/img_proof/tests/SLES/test_sles_hardened.py

```diff
@@ -0,0 +1,59 @@
+import pytest
+
+
+# Workaround for https://github.com/OpenSCAP/openscap/issues/1796
+def setup_swap(host, swap_file):
+    mem_total = int(host.run(
+        "cat /proc/meminfo | awk '$0 ~ /MemTotal:/ { print $2 }'"
+    ).stdout.strip())
+    if mem_total < 16e6:
+        for command in [
+            "sudo fallocate -l 4G {}".format(swap_file),
+            "sudo chmod 600 {}".format(swap_file),
+            "sudo mkswap {}".format(swap_file),
+            "sudo swapon {}".format(swap_file),
+        ]:
+            result = host.run(command)
+            if result.rc != 0:
+                raise RuntimeError(result.stderr.strip())
+
+
+def test_sles_hardened(host, get_release_value, is_sles_sap):
+    variant = get_release_value('VARIANT_ID') or ''
+
+    if "hardened" not in variant:
+        pytest.skip('Skipping non-hardened image')
+
+    swap_file = "/tmp/swap"
+    try:
+        setup_swap(host, swap_file)
+        xml_path = "pub/projects/security/oval/suse.linux.enterprise.15.xml"
+        # Downloaded file should have slashes replaced by hyphens
+        xml_file = xml_path.replace("/", "-")
+        oscap_dir = "/tmp/oscap"
+        oscap_file = "/usr/share/xml/scap/ssg/content/ssg-sle15-ds.xml"
+        oscap_profile = "pcs-hardening-sap" if is_sles_sap() else "pcs-hardening"
+
+        host.run("mkdir {}".format(oscap_dir))
+        host.run(
+            "curl -o- https://ftp.suse.com/{path}.gz | gunzip -c > {dir}/{file}".format(
+                path=xml_path,
+                file=xml_file,
+                dir=oscap_dir,
+            )
+        )
+
+        result = host.run(
+            "sudo oscap xccdf eval --local-files {dir} --profile {profile} {file}".format(
+                dir=oscap_dir,
+                file=oscap_file,
+                profile=oscap_profile,
+            )
+        )
+    except RuntimeError as exc:
```

Comment:
We'd want to catch Exception here to catch all exceptions.

Reply:
Fixed.